test/cluster/test_refresh.py

```diff
@@ -0,0 +1,113 @@
+from test.pylib.manager_client import ManagerClient
+from test.cluster.object_store.conftest import format_tuples
+from test.cluster.object_store.test_backup import topo, create_cluster, take_snapshot, create_dataset, compute_scope, check_data_is_back
+from test.cluster.util import wait_for_cql_and_get_hosts
+from test.pylib.rest_client import read_barrier
+from test.pylib.util import unique_name
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("topology_rf_validity", [
+        (topo(rf = 1, nodes = 3, racks = 1, dcs = 1), True),
+        (topo(rf = 3, nodes = 5, racks = 1, dcs = 1), False),
+        (topo(rf = 1, nodes = 4, racks = 2, dcs = 1), True),
+        (topo(rf = 3, nodes = 6, racks = 2, dcs = 1), False),
+        (topo(rf = 3, nodes = 6, racks = 3, dcs = 1), True),
+        (topo(rf = 2, nodes = 8, racks = 4, dcs = 2), True)
+    ])
+async def test_refresh_with_streaming_scopes(manager: ManagerClient, topology_rf_validity):
+    '''
+    Check that refreshing a cluster with stream scopes works
+
+    This test creates a cluster specified by the topology parameter above,
+    configurable number of nodes, tacks, datacenters, and replication factor.
+
+    It creates a dataset, takes a snapshot and copies the sstables of all nodes to a temporary
+    location. It then truncates the table so all sstables are gone, copies all the sstables into
+    each node's upload directory, and refreshes the nodes given the scope passed as the test parameter.
+
+    The test then performs two types of checks:
+    1) Check that the data is back in the table by getting all mutations from the nodes and checking
+    that a random sample of them contains the expected key and that they are replicated according to RF * DCS factor.
+    2) Check that the streaming communication between nodes is as expected according to the scope parameter of the test.
+    This stage parses the logs and checks that the data was streamed to nodes within the configured scope.
+    '''
+ 
+    topology, rf_rack_valid_keyspaces = topology_rf_validity
+
+    servers, host_ids = await create_cluster(topology, rf_rack_valid_keyspaces, manager, logger)
+
+    cql = manager.get_cql()
+
+    await manager.api.disable_tablet_balancing(servers[0].ip_addr)
+
+    ks = 'ks'
+    cf = 'cf'
+    _, keys, _ = create_dataset(manager, ks, cf, topology, logger)
+
+    _, sstables = await take_snapshot(ks, servers, manager, logger)
+
+    logger.info(f'Move sstables to tmp dir')
+    tmpdir = f'tmpbackup-{str(uuid.uuid4())}'
+    for s in servers:
+        workdir = await manager.server_get_workdir(s.server_id)
+        cf_dir = os.listdir(f'{workdir}/data/{ks}')[0]
+        tmpbackup = os.path.join(workdir, f'../{tmpdir}')
+        os.makedirs(tmpbackup, exist_ok=True)
+
+        snapshots_dir = os.path.join(f'{workdir}/data/{ks}', cf_dir, 'snapshots')
+        snapshots_dir = os.path.join(snapshots_dir, os.listdir(snapshots_dir)[0])
```

Comment:
The `os.listdir(snapshots_dir)[0])` can be replaced with the 0th return value from the take_snapshot() above ;)
Please, take this for the followup, no need to change now